src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -344,8 +350,18 @@ CHIP_ERROR DefaultICDClientStorage::SerializeToTlv(TLV::TLVWriter & writer, cons
     return writer.EndContainer(arrayType);
 }
 
+CHIP_ERROR DefaultICDClientStorage::CheckFabricExistence(FabricIndex fabricIndex)
+{
+    for (auto & fabric_idx : mFabricList)
+    {
+        VerifyOrReturnError(fabric_idx != fabricIndex, CHIP_NO_ERROR);
```

Comment:
I think this would be much simpler to understand as an actual `if` condition without the double-negatives.

Reply:
agree, done, thanks